lib/client.js

```diff
@@ -190,9 +190,9 @@ function MqttClient (streamBuilder, options) {
   })
 
   // resubscribe
-  this.on('connect', function () {
-    if (this.options.clean && Object.keys(this._subscribedTopics).length > 0) {
-      this.subscribe(this._subscribedTopics)
+  this.on('reconnect', function () {
```

Comment:
can you add a unit test that cover this case, so we check we don't get a duplicate subscription?